_posts/2022-01-19-#23201.md

```diff
@@ -10,14 +10,80 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+* Funding a transaction involves selecting inputs with a total amount that covers both the
+  payment(s) and the fees for the transaction itself. The target feerate is calculated at the
+  beginning, and the fees depend on the total size of the transaction.
+
+* In some use cases, users may want to fund a transaction with *external inputs*, i.e., inputs that
+  are not controlled by the user themselves, but some external wallet such as that of a LN
+  channel counterparty. The challenge in this scenario is accounting for these inputs when funding
+  the fees of the transaction - the wallet needs to know their size.
+
+    - For standard output types, the wallet can approximate the size of the input and corresponding
+      witness data. For example,
+[this stackexchange post](https://bitcoin.stackexchange.com/questions/48279/how-big-is-the-input-of-a-p2pkh-transaction)
+      breaks down the possible sizes of a P2PKH input.
+
+    - Other standard output types that involve scripts (which can be arbitrary in size), such as
+      P2SH, P2WSH and P2TR, may require the wallet to know more information.
+
+    - [PR #17211](https://github.com/bitcoin/bitcoin/pull/17211) added support for allowing
+      users to provide *solving data* including public keys, scripts, and descriptors to help the
+      wallet determine the size of external input(s).
+
+    - Some sizes might differ depending on how the external wallet represents the public key and
```

Comment:
The way the pubkey is represented is fixed by the UTXO via a hash. It's only the signature size that is unknown.